aspnetcore/aspnetcore-2.1.md

```diff
@@ -0,0 +1,188 @@
+
+## Integration tests
+
+A new package is introduced that streamlines test creation and execution. The [Microsoft.AspNetCore.Mvc.Testing](https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc.Testing/) package handles the following tasks:
+
+* Copies the dependencies file (*\*.deps*) from the tested app into the test project's *bin* folder.
+* Sets the content root to the tested app's project root so that static files and pages/views are found when the tests are executed.
+* Provides the [WebApplicationFactory](/dotnet/api/microsoft.aspnetcore.mvc.testing.webapplicationfactory-1) class to streamline bootstrapping the tested app with [TestServer](/dotnet/api/microsoft.aspnetcore.testhost.testserver).
+
+The following test uses [xUnit](https://xunit.github.io/) to check that the Index page loads with a success status code and with the correct Content-Type header:
+
+```csharp
+public class BasicTests 
+    : IClassFixture<WebApplicationFactory<RazorPagesProject.Startup>>
+{
+    private readonly HttpClient _client;
+        
+    public BasicTests(WebApplicationFactory<RazorPagesProject.Startup> factory)
+    {
+        _client = factory.CreateClient();
+    }
+
+    [Fact]
+    public async Task GetHomePage()
+    {
+        // Act
+        var response = await _client.GetAsync("/");
+
+        // Assert
+        response.EnsureSuccessStatusCode(); // Status Code 200-299
+        Assert.Equal("text/html; charset=utf-8", 
+            response.Content.Headers.ContentType.ToString());
+    }
+}
+```
+
+For more information, see the [Integration tests](xref:test/integration-tests) topic.
+
+## [ApiController], ActionResult
+
+In ASP.NET Core 2.1, improved support for OpenAPI specification(previously known as **Swagger**)) has been added to Web API. `ActionResult<T>` is a new type added to allow an application to return either a response type or any other action result (similar to IActionResult), while still indicating the response type. The `[ApiController] attribute has also been added as the way to opt-in to Web API-specific conventions and behaviors.
+
+For more information, see [Build Web APIs with ASP.NET Core](xref:web-api/index)
+
+## IHttpClientFactory
+
+The new `HttpClientFactory` type can be registered and used to configure and consume instances of HttpClient in applications. HttpClient already has the concept of delegating handlers that could be linked together for outgoing HTTP requests. The factory will make registering of these per named client more intuitive as well as implement a Polly handler that allows Polly policies to be used for Retry, CircuitBreakers, etc.
```

Comment:
`HttpClientFactory` is actually an existing type in the System.Net namespace.

~~The new `HttpClientFactory` type can be registered and used~~ ASP.NET Core 2.1 includes a new `IHttpClientFactory` service that makes it easier

Reply:
will make -> makes